cpp/include/cudf/detail/copy_if_else.cuh

```diff
@@ -34,47 +34,36 @@ namespace detail {
 
 namespace {    // anonymous
 
-template <typename T, typename Filter, bool has_validity>
+template <size_type block_size, typename T, typename Filter, bool has_validity>
 __global__
```

Comment:
Since you are templating block_size, you can use __launch_bounds__ to help the  compiler optimize: `__launch_bounds__(block_size)` after the `template <...>` before `__global__`.